python/tvm/rpc/base.py

```diff
@@ -136,13 +139,30 @@ def random_key(prefix, cmap=None):
     key : str
         The generated random key
     """
-    if cmap:
-        while True:
-            key = prefix + str(random.random())
-            if key not in cmap:
-                return key
-    else:
-        return prefix + str(random.random())
+    while True:
+        key = "{}{}{}".format(prefix, delimiter, random.random())
+        if not cmap or key not in cmap:
+            break
```

Comment:
Is it expected that if the `key` is found in `cmap` then on the next iteration it contains more delimiters than only one?
E.g.:
- First iteration:
  ```
  prefix = "127.0.0.1:5555"
  key = "127.0.0.1:5555:123456"
  ```
  `cmap` has already contained this key `127.0.0.1:5555:123456`. We will iterate once again.
- Second iteration:
  ```
  key = "127.0.0.1:5555:123456:654321"
  ```
  
If we split this key after the second iteration, then we get `127.0.0.1:5555:123456` but it is not the key.
Probably this change will help us to avoid this problem?
```suggestion
        key = "{}{}{}".format(prefix, delimiter, random.random())
        if not cmap or key not in cmap:
            break
        delimiter = ""
```
Or maybe I missed something and everything should work?

Reply:
there is no saving of the variable key between iterations of the loop. there is a test for this situation. tests/python/unittest/test_rpc_base.py

Reply:
Ups, sorry, my bad. Didn't notice that. Thank you.